File: src/python/_1493_Longest_Subarray_of_1's_After_Deleting_One_Element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l = 0
        r = 0
        zeroIndex = -1
        ans = 0

        while r < len(nums):
            if nums[r] == 0:
                ans = max(ans, r - l - 1)
                if zeroIndex >= 0:
                    l = zeroIndex + 1
                zeroIndex = r
            r += 1

        ans = max(ans, r - l - 1)

        if zeroIndex < 0:
            return len(nums) - 1
        else:
            return ans

File: src/_1493_Longest_Subarray_of_1's_After_Deleting_One_Element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int l = 0, r = 0, zeroIndex = -1, ans = 0;

        while (r < nums.length) {
            if (nums[r] == 0) {
                ans = Math.max(ans, r - l - 1);
                if (zeroIndex >= 0)
                    l = zeroIndex + 1;
                zeroIndex = r;
            }
            r += 1;
        }

        ans = Math.max(ans, r - l - 1);

        if (zeroIndex < 0)
            return nums.length - 1;

        return ans;
    }
}
